doc/versioning_and_life_cycle.md

```diff
@@ -0,0 +1,73 @@
+
+## Versioning
+
+The Azure CLI follows [Semantic Versioning](https://semver.org/) for version numbering. The CLI version consists of three parts `MAJOR.MINOR.PATCH`:
+- MAJOR version is upgraded only when there is a huge revolution, such as the core changes that will break commands' behavior globally.
```

Comment:
Because the actual situation of CLI and the definition of `Semantic Versioning` are not exactly the same, For example, CLI upragde different versions for normal bugfix and special hotfix, and the major version has only been upgraded once in the history, which are different from `Semantic Versioning`.

Reply:
I'd like to make the versioning description more suitable for the actual situation of CLI. Do you think I need to change `The Azure CLI follows Semantic Versioning for version numbering` to `The Azure CLI refers to Semantic Versioning, but has it's unique rules for version numbering`?

Reply:
I think that either the tool follows semver or not. If it does not, then you need to explain the logic behind each version change and argument with examples.